coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -631,14 +633,59 @@ func (server *Server) CompleteJob(ctx context.Context, completed *proto.Complete
 			if err != nil {
 				return xerrors.Errorf("update workspace build: %w", err)
 			}
+
+			agentTimeouts := make(map[time.Duration]bool) // A set of agent timeouts.
 			// This could be a bulk insert to improve performance.
 			for _, protoResource := range jobType.WorkspaceBuild.Resources {
+				for _, protoAgent := range protoResource.Agents {
+					dur := time.Duration(protoAgent.GetConnectionTimeoutSeconds()) * time.Second
```

Comment:
review: should we truncate these to `time.Second`?

Reply:
I don't think that's necessary since the input is in seconds?